scraper/scraper.ts

```diff
@@ -1,28 +1,974 @@
+  if (!(career && Object.values(Career).includes(career))) {
+    throw new Error('Invalid Career: ' + career)
+  }
+  index++
+
+  const termsOffered: string[] = []
+  const censusDates: string[] = []
+  let notes: string[] = []
+  // Find all the terms the course runs in
+  while (index < data.length) {
+    // Term regex -> letter, letter or number, optional letter or number
+    const termFinderRegex = /^[A-Z][A-Z0-9][A-Z0-9]?$/
+    if (termFinderRegex.test(data[index])) {
+      termsOffered.push(data[index])
+    }
+
+    // Look for the census dates in case there is no class to extract dates from
+    const censusDateFinder = /^[0-9]+-[A-Z]+-[0-9]+$/
+    if (censusDateFinder.test(data[index])) {
+      censusDates.push(data[index])
+      notes.push(data[index + 1])
+    }
+    index++
+  }
+
+  const courseData: CourseInfo = {
+    school: school,
+    campus: campus,
+    career: career,
+    termsOffered: termsOffered,
+    censusDates: censusDates,
+  }
+
+  return { notes: notes, courseInfo: courseData }
+}
+
+/**
+ * Parses data from the data array into a class object
+ * @param data: array of text from elements with a data class
+ * from a class chunk
+ */
+const parseClassChunk = (
+  data: Chunk
+): { classData: Class; warnings: classWarnings[] } | false => {
+  let index = 0
+  const warnings: classWarnings[] = []
+
+  // ClassID is a 4 or 5 digit number
+  const classID = parseInt(data[index])
+  const classIDChecker = /^[0-9]{4,5}$/
+  if (!classIDChecker.test(data[index])) {
+    throw new Error('Invalid Class Id: ' + classID)
+  }
+  index++
+
+  // Section is an 4 character alphanumeric code
+  const section = data[index]
+  const sectionChecker = /^[A-Z0-9]{0,4}$/
+  if (!sectionChecker.test(section)) {
+    throw new Error('Invalid Section: ' + section)
+  }
+  index++
+
+  let term: string
+  const termFinderRegex = /^([A-Z][A-Z0-9][A-Z0-9]?).*/
+  if (termFinderRegex.test(data[index])) {
+    term = termFinderRegex.exec(data[index])[1]
+  }
+  index++
+
+  // Check if the class is actually course enrolment
+  // which is not needed
+  const courseEnrolmentChecker = /[Cc]ourse [Ee]nrolment/
+  if (courseEnrolmentChecker.test(data[index])) {
+    // Abort
+    return false
+  }
+
+  const activity = data[index]
+  if (!activity) {
+    throw new Error('Unknown activity: ' + activity)
+  }
+  index++
+
+  const status: Status = <Status>data[index]
+  if (!Object.values(Status).includes(status)) {
+    throw new Error('Invalid Status: ' + status)
+  }
+  index++
+
+  const enrAndCap = data[index].split('/')
+  const courseEnrolment = {
+    enrolments: parseInt(enrAndCap[0]),
+    capacity: parseInt(enrAndCap[1]),
+  }
+  // Enrolments and capacity are both numbers and enrolment less than capacity
+  // (Strict requirement)
+  if (
+    !courseEnrolment ||
+    !(courseEnrolment.enrolments >= 0 && courseEnrolment.capacity > 0) ||
+    courseEnrolment.enrolments > courseEnrolment.capacity
```

Comment:
Can reduce this entire if condition to one condition
1. (line 490) - courseEnrolment is always defined (defined right above)
2. (line 491 and 492) - 492 negates the need to check in 491 if they are both 0, if we use the exclusive greater than comparator (>)  
```suggestion
    courseEnrolment.enrolments > courseEnrolment.capacity
```

Reply:
Yeah lol